### ISSUE TITLE
Parallelism with `--hosts`

### ISSUE THREAD

**Issue by lkdmid** [fabric/fabric#2156]
Greetings,

Fabric 2.5.0
Paramiko 2.7.2
Invoke 1.4.1

_**TL;DR:** I can't see how to execute a task concurrently when using `--hosts`._

Given a list of hosts we can obviously create a `ThreadingGroup()` object (e.g. `target_group = ThreadingGroup(*hosts)`) and it works as expected, but I'm struggling to find **any way** to achieve this when using the `fab --hosts <host1>,<host2> ...` CLI syntax, which is making my life difficult as I'm maintaining an ad-hoc tool which demands the CLI entry of target hosts.

Even just optionally making the `--hosts` argument available in the form of a list would allow us to pass it to `ThreadingGroup()` (given, we'd then need to prevent fab repeating the run for each passed hostname), but it doesn't, we just get a single `Connection` object while `fab` runs trough them in serial.

```python
$ fab -H 1,2,3,4,5 test-concurrent
<Connection host=1> ... then 2 ... then 3 ... etc
```

Unless I'm missing something, this sort of seems like missing basic functionality.

Any suggestions appreciated.

**Comment by davidjmemmett** [comment]
Hi @lkdmid,

[Global parallel execution has not yet been implemented](https://github.com/pyinvoke/invoke/issues/63) in fabric v2/invoke. If you are able to parse the arguments yourself and create your own `ThreadingGroup`s, you ought to be able to split the hosts and pass them though, unless I'm missing something.

Cheers,
David

**Comment by lkdmid** [comment]
Hi @davidjmemmett - thanks for the reply.

Yup - I eventually came to the conclusion that dropping the `fab` CLI and implementing my own `argparse` was the only realistic way to go.

Onwards and upwards.
